app/jobs/update_github_team_job.rb

```diff
@@ -0,0 +1,11 @@
+class UpdateGithubTeamJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(user_id, team_id, options)
```

Comment:
You can pass a user object with ActiveJob and remove the guard for a nil user: http://edgeguides.rubyonrails.org/active_job_basics.html#globalid